File: src/main/java/com/group8/ClassroomPal/model/Assignment.java
package com.group8.ClassroomPal.model;

public class Assignment {

    public String uid;
    public String course_uid;
    public String title;
    public String content_type;
    public String content_data;

    public Assignment() {}

    public Assignment(String uid, String course_uid, String title,
                      String content_type, String content_data) {
        this.uid = uid;
        this.course_uid = course_uid;
        this.title = title;
        this.content_type = content_type;
        this.content_data = content_data;
    }
}

File: src/main/resources/frontend/submission/assignment.js
async function publishAssignment(){

    const assignment={

        course_uid:getQuery("course_uid"),

        title:document.getElementById("title").value,

        content_type:"TEXT",

        content_data:document.getElementById("content").value

    };

    const result=await apiPost(

        API.ASSIGNMENT,

        assignment

    );

    alert(result.status);

}

async function loadAssignments(){

    const courseUid=getQuery("course_uid");

    const list=await apiGet(

        API.ASSIGNMENT+"/course/"+courseUid

    );

    render(list);

}

function render(assignments){

    const div=document.getElementById("assignmentList");

    div.innerHTML="";

    assignments.forEach(a=>{

        div.innerHTML+=`

            <p>

                ${a.title}

            </p>

        `;

    });

}